controllers/secret.go

```diff
@@ -24,187 +24,108 @@ import (
-func (r *OperatorPipelineReconciler) reconcilePyxisAPISecret(meta metav1.ObjectMeta) error {
-	pipelineName := types.NamespacedName{
-		Namespace: meta.Namespace,
-		Name:      meta.Name,
-	}
-	pipeline := &certv1alpha1.OperatorPipeline{}
-	err := r.Client.Get(context.TODO(), pipelineName, pipeline)
+func (r *OperatorPipelineReconciler) ensurePyxisAPISecret(meta metav1.ObjectMeta) error {
+	operatorPipeline, err := r.getPipeline(meta)
 	if err != nil {
-		if k8serrors.IsNotFound(err) {
-			log.Info("pipeline resource not found. Ignoring since object must be deleted")
-			return nil
-		}
-		log.Error(err, "unable to retrieve pipeline resource in namespace "+meta.Namespace)
+		log.Error(err, "unable to resolve pyxis secret for %s in %s", meta.Name, meta.Namespace)
 		return err
 	}
-	var secretName = PYXIS_API_SECRET
-	if pipeline.Spec.PyxisApiSecretName != "" {
-		secretName = pipeline.Spec.PyxisApiSecretName
-	}
-	key := types.NamespacedName{
-		Namespace: meta.Namespace,
-		Name:      secretName,
+	secretName := PYXIS_API_SECRET_NAME
+	if operatorPipeline.Spec.PyxisSecretName != "" {
+		secretName = operatorPipeline.Spec.PyxisSecretName
 	}
-	secret := newSecret(key)
-	if !IsObjectFound(r.Client, key, secret) {
+	return r.ensureSecret(secretName, PYXIS_API_KEY_NAME, meta)
+}
+
+func (r *OperatorPipelineReconciler) ensureSecret(secretName string, secretKeyName string, meta metav1.ObjectMeta) error {
+	namespacedSecretName := newNamespacedName(secretName, meta.Namespace)
+	secret := &corev1.Secret{}
+	if !IsObjectFound(r.Client, namespacedSecretName, secret) {
 		err := errors.New(fmt.Sprintf("an existing secret named %s was not found in namespace %s", secretName, meta.Namespace))
 		log.Error(err, fmt.Sprintf("unable to reconcile %s in namespace %s", secretName, meta.Namespace))
 		return err
 	}
-	log.Info(fmt.Sprintf("existing %s secret found in namespace %s", secretName, meta.Namespace))
-	pyxisApiSecret := &corev1.Secret{}
-	if r.Client.Get(context.TODO(), key, pyxisApiSecret) != nil {
-		log.Error(err, fmt.Sprintf("unable to get the %s secret", secretName))
+	log.Info(fmt.Sprintf("an existing secret named %s was found in namespace %s", secretName, meta.Namespace))
+	secret = &corev1.Secret{}
+	err := r.Client.Get(context.TODO(), namespacedSecretName, secret)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("unable to get the %s secret in the %s namespace", secretName, meta.Namespace))
 		return err
 	}
-	if pyxisApiSecret.Data[PYXIS_API_KEY] == nil || len(pyxisApiSecret.Data[PYXIS_API_KEY]) == 0 {
-		err = errors.New(fmt.Sprintf("the pyxis api key in %s is empty!", secretName))
-		log.Error(err, fmt.Sprintf("The %s secret does not contain the expected key", secretName))
+	log.Info(fmt.Sprintf("successfully fetched secret named %s in namespace %s", secretName, meta.Namespace))
+	err = errors.New(fmt.Sprintf("the %s secret does not contain a valid key %s", secretName, secretKeyName))
```

Comment:
I prefer keeping the granularity that the original errors in the `git hub` code so we can help trouble shoot easier. If others feel that a single error for both cases is also okay, I'm okay with that as well.

Reply:
think they are equivalent.. both would lead to checking the value of the secret and discovering that its either not present or has a bad value..

Reply:
regarding the errors file.. seems like this would be the only place we would use those errors no ? still better to separate ?

Reply:
It seems like we use this error twice, and seeing an error declaration in the middle of the code is a bit confusing. You could keep the error in errors.go generic, and use the error wrapping to add more context (e.g. the secret name, etc)

Reply:
the secret errors would likely only be used here but it just gives us a central location to manage all of the errors. i think that's the purpose

Reply:
yeah defining new errors in the middle is indeed jarring will start the errors file can add stuff in as we go 👍